fe/fe-core/src/main/java/com/starrocks/catalog/HudiTable.java

```diff
@@ -0,0 +1,391 @@
+
+    private List<String> partColumnNames = Lists.newArrayList();
+    // dataColumnNames stores all the non-partition columns of the hive table,
+    // consistent with the order defined in the hive table
+    private List<String> dataColumnNames = Lists.newArrayList();
+
+    private Map<String, String> hudiProperties = Maps.newHashMap();
+
+    public HudiTable() {
+        super(TableType.HUDI);
+    }
+
+    public HudiTable(long id, String name, List<Column> schema, Map<String, String> properties) throws DdlException {
+        super(id, name, TableType.HUDI, schema);
+        validate(properties);
+    }
+
+    public String getDb() {
+        return db;
+    }
+
+    public String getTable() {
+        return table;
+    }
+
+    public List<Column> getPartitionColumns() {
+        List<Column> partColumns = Lists.newArrayList();
+        for (String columnName : partColumnNames) {
+            partColumns.add(nameToColumn.get(columnName));
+        }
+        return partColumns;
+    }
+
+    public List<String> getPartitionColumnNames() {
+        return partColumnNames;
+    }
+
+    public List<String> getDataColumnNames() {
+        return dataColumnNames;
+    }
+
+    public Map<PartitionKey, Long> getPartitionKeys() throws DdlException {
+        List<Column> partColumns = getPartitionColumns();
+        return Catalog.getCurrentCatalog().getHiveRepository()
+                .getPartitionKeys(resourceName, db, table, partColumns);
+    }
+
+    public String getResourceName() {
+        return resourceName;
+    }
+
+    public HoodieTableType getTableType() {
+        return HoodieTableType.valueOf(hudiProperties.get(HUDI_TABLE_TYPE));
+    }
+
+    public String getHudiHiveMetastoreUris() {
+        return hudiProperties.get(HUDI_METASTORE_URIS);
+    }
+
+    public String getHudiBasePath() {
+        return hudiProperties.get(HUDI_BASE_PATH);
+    }
+
+    public String getHudiTableInputFormat() {
+        return hudiProperties.get(HUDI_TABLE_INPUT_FORMAT);
+    }
+
+    public String getHudiMatadataEnable() {
+        return hudiProperties.get(HUDI_MATADATA_ENABLE);
+    }
+
+    private void validate(Map<String, String> properties) throws DdlException {
+        if (properties == null) {
+            throw new DdlException("Please set properties of iceberg table, they are: database, table.");
+        }
+
+        Map<String, String> copiedProps = Maps.newHashMap(properties);
+        this.db = copiedProps.get(HUDI_DB);
+        if (Strings.isNullOrEmpty(db)) {
+            throw new DdlException(String.format(PROPERTY_MISSING_MSG, HUDI_DB, HUDI_DB));
+        }
+        copiedProps.remove(HUDI_DB);
+
+        this.table = copiedProps.get(HUDI_TABLE);
+        if (Strings.isNullOrEmpty(table)) {
+            throw new DdlException(String.format(PROPERTY_MISSING_MSG, HUDI_TABLE, HUDI_TABLE));
+        }
+        copiedProps.remove(HUDI_TABLE);
+
+        String hudiMetaDataEnable = copiedProps.get(HUDI_MATADATA_ENABLE);
```

Comment:
what does HUDI_MATADATA_ENABLE used for?

Reply:
https://hudi.apache.org/docs/metadata

removed, I will support this feature in the following PR.